### PR TITLE
Introducing some limits to avoid high memory consumption in corner cases

### PR DESCRIPTION
**Pull request by ivanprado**
Text and css generated features are unlimited, and they can be very big in some cases with a lot of text or nodes. Also, some pages have a very high number of links. This PR imposes some limits over them to avoid excessive memory consumption in such a case. 

Example of pages with such memory consumption (especially the first one): 
- https://www.lego.com/de-de/kids 
- https://www.cotswoldoutdoor.com/c/sale/shop-all.html
- https://www.nobodyschild.com/clothing.html

Based on the stats below and some runs, these are the default limits:

* max_text_features=80
* max_css_features=120
* max_links=2500

```
Stats for number of links (run over ~3000 pages):

             links
count  2986.000000
mean    331.445747
std     363.960655
min       0.000000
50%     237.500000
75%     403.000000
90%     664.000000
95%     899.750000
98%    1277.600000
99%    1602.600000
max    6592.000000

Stats for number of text and css features:

                text        classes
count  989697.000000  989697.000000
mean       35.574637      43.741996
std       206.472999      78.599356
min         0.000000       0.000000
50%        26.000000      22.000000
75%        42.000000      58.000000
90%        64.000000     108.000000
95%        84.000000     150.000000
98%       122.000000     226.000000
99%       158.000000     310.000000
max     65507.000000    5863.000000
```

Metrics. Baseline (copied from the notebook in master):
```py

             precision    recall  f1-score   support

       PAGE      0.871     0.924     0.897      1178
       NEXT      0.881     0.768     0.821       155
       PREV      0.897     0.777     0.833       112

avg / total      0.874     0.896     0.884      1445

Sequence accuracy: 0.582
CPU times: user 9.27 s, sys: 1.4 s, total: 10.7 s
Wall time: 1min 6s
```

New metrics:
```py

              precision    recall  f1-score   support

        PAGE      0.872     0.925     0.898      1178
        NEXT      0.903     0.781     0.837       155
        PREV      0.892     0.812     0.850       112

   micro avg      0.876     0.901     0.888      1445
   macro avg      0.889     0.839     0.862      1445
weighted avg      0.877     0.901     0.888      1445

Sequence accuracy: 0.612
CPU times: user 1min 4s, sys: 1.56 s, total: 1min 5s
Wall time: 1min 20s
```

Besides, this PR contains some changes already on Memex upstream: scikity 0.18 and eli5.

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -2,7 +2,7 @@
 ; under all supported Python interpreters
 
 [tox]
-envlist = py27,py33,py34,py35
+envlist = py27,py33,py38
```
Comment (kmike) [inline tox.ini:5]: do you want to keep py33 here? it is dropped in travis.yml

Comment (ivanprado) [reply tox.ini:5]: It seems travis is not working in this repository. But in the fork, it is enabled, and even if Travis is configured to work with Python 3.8, Python 3.5 is used by Tox. See this run: https://github.com/scrapinghub/autopager/pull/1/checks?check_run_id=1081133825

I could remove python 3.5 or even python 2.7, but why not leave them given that they are working?

**Comment by kmike** [comment]
Nice, thanks @ivanprado!
